### ISSUE TITLE
Unable to compile, macOS High Sierra 10.13.4

### ISSUE THREAD

**Issue by napulen** [humdrum-tools/humdrum-tools#1]
I am getting the submodules with the `--recursive` flag.

Running `make` on the root of humdrum-tools, I am getting this error
```
error: use of undeclared identifier 'smf'
```
It repeats for several files until it ends with a
```
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
```
Am I missing something?

**Comment by napulen** [comment]
The weirdest thing is that the "max-errors" flag does not seem to be set anywhere in the makefiles of humdrum-tools, and the default option for that is 0 (unlimited errors before stopping the compilation), so I have no idea how is this error limit triggering. Maybe something related to Xcode somewhere?

**Comment by craigsapp** [comment]
I am fixing the problem, and it should be ready soon 😛   The short answer is that I added the `smf` namespace to the midifile library which humextra uses, but I did not update the humextra programs that use it.

**Comment by craigsapp** [comment]
The source of the problem is that I added the `smf` namespace to the midifile library:
     https://github.com/craigsapp/midifile

This library is used by `mid2hum`, `hum2mid` `mkeyscape` and `humplay` which I had to update to use the new namespace.  I already had humextra installed and those programs compiled, so since they were not changed and the Makefile did not check that the midifile was updated, they were not recompiled (so I didn't notice the problem myself).

I can now do a fresh install and it compiles successfully:

```bash
git clone --recursive https://github.com/humdrum-tools/humdrum-tools
cd humdrum-tools
make update  # not necessary but good to do
make
```

It should also work to update an existing installation:

```bash
make update
make
```

If that does not work (due to a reconfiguration of the libraries most likely), then try the fresh install method).
